02-express-tutorial/controllers/people.js

```diff
@@ -0,0 +1,59 @@
+const {people} = require('../data')
+
+const getPeople = (req, res) => {
+    res.status(200).json({ sucess: true, data: people})
+}
+const addPerson = (req, res) =>{
+    const {name} = req.body
+    if (name) {
+        people.push({ id: people.length + 1, name: name });
+        res.status(201).json({ success: true, name: name});
+    }
+    res.status(400).json({ success: false, message: "Please provide a name" });
```

Comment:
This line is always executed and causes an error on the server side
Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client

Refactoring the code to check if name is missing first would resolve this issue
```
if (!name) {
 res.status(400).json({ success: false, message: "Please provide a name" });
}
 
people.push({ id: people.length + 1, name: name });
res.status(201).json({ success: true, name: name});
```